### ISSUE TITLE
Option to restrict to or default to logging into certain instance

### ISSUE THREAD

**Issue by cassidyjames** [cheeaun/phanpy#284]
I'm hosting Phanpy on a subdomain of my Mastodon instance to offer it as an alternate front-end. While there's not really a technical reason to restrict it to users of my instance, experience-wise it would be nice if I could easily scope it to that instance and not ask users what their instance is.

I can think a bit more about the experience/flow that would be ideal if you're curious.

**Comment by cheeaun** [comment]
Would be interesting to know what you think!

Few notes and questions, on my side, as of now:
- If I'm not mistaken, [Elk](https://github.com/elk-zone/elk) has this feature. Will need to look at it.
- If Phanpy is used as alternate front-end, it means the official front-end will still be around?
  - Phanpy doesn't implement *everything* yet, so official front-end is still needed.
  - There's no account registration on Phanpy. It can be done, but will need some time/effort.
  - Easy way would be linking "Log in" button on Phanpy straight to the instance's official front-end auth page.
- After user log in to (one) instance, can they still add account from other instances? Or should it be restricted/hidden?
- Once restricted/customized to a single instance, should the app icon (and title) change?
  - Phanpy is a PWA, so it can be installed on desktop/mobile.
  - Icon can probably be the instance's logo, title could be the instance domain/name.
  - Even if not installed, favicon change would be useful too, so that users can differentiate the tabs.

**Comment by cassidyjames** [comment]
Yeah I don't think I'd see this as a complete replacement for the Mastodon front-end (at least not yet!); I'd still have it live at its own domain, even. But it could be nice as an alternate front-end that doesn't require manually typing in the instance name; you could just have a `Log In` button that'd take you to https://example.com/oauth/authorize instead of having that step that asks for the instance name. 😄

**Comment by cassidyjames** [comment]
D'oh! It helps if I actually read and comprehend your reply. Let me respond to that, now. :)

>Easy way would be linking "Log in" button on Phanpy straight to the instance's official front-end auth page.

Yep, 100%. This would get me like 99% of the way there, I think.

>After user log in to (one) instance, can they still add account from other instances? Or should it be restricted/hidden?

Good question; I guess if the implication was that you're scoped to one instance, then you'd not allow logging into other instances here, either. You _could_ still have the option to add an account which would take them to the oauth page again, where they could log out from the Mastodon front-end, then log in with another account, then finish the authentication... but that does seem kinda messy. Maybe it'd be best to just support single accounts in this "mode".

>Once restricted/customized to a single instance, should the app icon (and title) change?

Oh yeah good question; the way the Mastodon front-end does it is it replaces the PWA title with the instance's name.

>Even if not installed, favicon change would be useful too, so that users can differentiate the tabs.

Yeah good point. Then again, Mastodon itself _doesn't_ do this, so not a huge deal if Phanpy didn't.

**Comment by cheeaun** [comment]
> > After user log in to (one) instance, can they still add account from other instances? Or should it be restricted/hidden?
> 
> Good question; I guess if the implication was that you're scoped to one instance, then you'd not allow logging into other instances here, either. You _could_ still have the option to add an account which would take them to the oauth page again, where they could log out from the Mastodon front-end, then log in with another account on the same instance, then finish the authentication... but that does seem kinda messy. Maybe it'd be best to just support single accounts in this "mode".

Yeah, it does seem messy.
But one perhaps rare/edge case would be a user with **multiple** accounts on the *same* instance (I honestly don't test this at all, there're probably bugs somewhere 🤪)

For the app + icons etc, it's probably possible by providing extra variables or args when building, could be something like `VITE_WEBSITE=https://phanpy.blaede.family VITE_INSTANCE_NAME="Blaede Family" npm run build`. Icons need more… effort (various dimensions, etc) so could be left out first 😆
